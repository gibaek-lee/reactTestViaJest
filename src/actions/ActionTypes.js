/* Declare action types */

//of App.js
export const INSERT = 'INSERT';

//of Counter.js
export const INCREASE = 'INCREASE';
export const DECREASE = 'DECREASE';

//of NameForm.js
export const SUBMIT = 'SUBMIT';
export const CHANGE = 'CHANGE';
