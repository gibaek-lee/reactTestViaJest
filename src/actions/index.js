/* Declare action creators */

import * as types from './ActionTypes';

export const increase = () => ({
  type: types.INCREASE
});

export const decrease = () => ({
  type: types.DECREASE
});

export const change = (name) => ({//store의 action parameter(action.name)
  type: types.CHANGE,
  name: name
});

export const insert = () => ({
  type: types.INSERT
});

export const submit = () => ({
  type: types.SUBMIT
});
