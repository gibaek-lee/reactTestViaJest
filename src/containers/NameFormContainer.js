import NameForm from '../components/NameForm';
import * as actions from '../actions';
import { connect } from 'react-redux';

const mapStateToProps = (state) => ({
  name: state.nameData.name
});

const mapDispatchToProps = (dispatch) => ({
  onChange: (name) => dispatch(actions.change(name)),
  onInsert: () => dispatch(actions.insert()),
  onSubmit: () => dispatch(actions.submit())
});

const NameFormContainer = connect(
  mapStateToProps, mapDispatchToProps
)(NameForm);

export default NameFormContainer;
