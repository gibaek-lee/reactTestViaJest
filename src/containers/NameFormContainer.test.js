import React from 'react';
import { mount } from 'enzyme';//enzyme full rendering
import NameFormContainer from './NameFormContainer';
import configureMockStore from 'redux-mock-store';
import * as nameActions from '../actions/index';

describe('NameFormContainer', () => {
  let component = null;
  const mockStore = configureMockStore();

  let store = mockStore({
    nameData: {
      name: '',
      nameList: []
    }
  });

  it('renders properly', () => {
    component = mount(<NameFormContainer store={store}/>)
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('dispatches CHANGE action', () => {
    const mockedEvent = {
      target: {
        value: undefined//change action creator의 name은 undefined이다.
      }
    };
    component.find('input').simulate('change',mockedEvent);
    expect(store.getActions()[0]).toEqual(nameActions.change());
  });

  it('dispatches INSERT/SUMBIT action', () => {
    const mockedEvent = {
      preventDefault: () => null
    }
    component.find('form').simulate('submit',mockedEvent);
    expect(store.getActions()[1]).toEqual(nameActions.insert());
    expect(store.getActions()[2]).toEqual(nameActions.submit());
    //console.log(store.getActions());
  });
});
