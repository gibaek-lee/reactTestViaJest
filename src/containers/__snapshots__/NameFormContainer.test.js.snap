// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameFormContainer matches snapshot 1`] = `
<Connect(NameForm)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <NameForm
    name=""
    onChange={[Function]}
    onInsert={[Function]}
    onSubmit={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <form
      onSubmit={[Function]}
    >
      <label>
        Your name
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </NameForm>
</Connect(NameForm)>
`;
